packages/AdaptiveCards/Actions/SendActionExecuteResponse.cs

```diff
@@ -0,0 +1,125 @@
+        public IntExpression StatusCode { get; set; } = new IntExpression(200);
+
+       [JsonConverter(typeof(StringEnumConverter), /*camelCase*/ true)]
+        public enum ValueType
+        {
+            /// <summary>
+            /// The response includes an Adaptive Card
+            /// </summary>
+            AdaptiveCard,
+
+            /// <summary>
+            /// The response includes a message
+            /// </summary>
+            Message
+        }
+
+        [JsonProperty("type")]
+        public EnumExpression<ValueType> Type { get; set; } = new EnumExpression<ValueType>(ValueType.Message);
+
+        [JsonProperty("value")]
+        public ValueExpression Value { get; set; }
+
+        public async override Task<DialogTurnResult> BeginDialogAsync(DialogContext dc, object options = null, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            if (options is CancellationToken)
+            {
+                throw new ArgumentException($"{nameof(options)} cannot be a cancellation token");
+            }
+
+            if (this.Disabled != null && this.Disabled.GetValue(dc.State) == true)
+            {
+                return await dc.EndDialogAsync(cancellationToken: cancellationToken).ConfigureAwait(false);
+            }
+
+            if (dc.Context.Activity.Type != ActivityTypes.Invoke || dc.Context.Activity.Name != "adaptiveCard/action")
+            {
+                throw new Exception($"{this.Id}: should only be called in repsonse to 'invoke' activities with a name of 'adaptiveCard/action'.");
+            }
+
+            // Get parameters
+            var statusCode = StatusCode?.GetValue(dc.State);
+            var type = Type?.GetValue(dc.State);
+            var value = Value?.GetValue(dc.State);
+            string mimeType;
+
+            // Validate params
+            if (statusCode != null && statusCode >= 200 && statusCode < 300)
+            {
+                switch (type)
+                {
+                    case ValueType.Message:
+                        mimeType = "application/vnd.microsoft.activity.message";
+                        if (!(value is String))
+                        {
+                            throw new ArgumentException($"{this.Id}: Message 'value' isn't of type String.");
+                        }
+                        break;
+                    case ValueType.AdaptiveCard:
+                        mimeType = "application/vnd.microsoft.card.adaptive";
+                        if (!(value is JObject))
+                        {
+                            throw new ArgumentException($"{this.Id}: Card 'value' isn't of type Object.");
+                        }
+                        break;
+                    default:
+                        throw new ArgumentException($"{this.Id}: A 'type' wasn't specified.");
+                }
+
+            }
+            else
+            {
+                value = null;
```

Comment:
If you initialize these variables with these values and then move `value = this.Value?.GetValue(dc.State);` into the true branch, then you can delete the else branch.